packages/plainjs/dist/deno/tokenizer.ts

```diff
@@ -441,24 +446,24 @@ export default class Tokenizer {
                   //<55296,56319> - highSurrogate
                   this.highSurrogate = intVal;
                 } else {
-                  this.bufferedString.appendBuf(
-                    this.encoder.encode(String.fromCharCode(intVal)),
-                  );
+                  const buf = this.encoder.encode(String.fromCharCode(intVal));
```

Comment:
I think that we can reduce the redundancies in this code block

```js
              const intVal = parseInt(
                this.unicode + String.fromCharCode(n),
                16,
              );
              let unicodeString: string;
              if (this.highSurrogate === undefined) {
                if (intVal >= 0xd800 && intVal <= 0xdbff) {
                  //<55296,56319> - highSurrogate
                  this.highSurrogate = intVal;
                  this.state = TokenizerStates.STRING_DEFAULT;
                  continue;
                } else {
                  unicodeString = String.fromCharCode(intVal);
                }
              } else {
                if (intVal >= 0xdc00 && intVal <= 0xdfff) {
                  //<56320,57343> - lowSurrogate
                  unicodeString = String.fromCharCode(this.highSurrogate, intVal);
                } else {
                  unicodeString = String.fromCharCode(this.highSurrogate);
                }
                this.highSurrogate = undefined;
              }
              const unicodeBuffer = this.encoder.encode(unicodeString);
              this.bufferedString.appendBuf(unicodeBuffer);
              this.escapedCharsByteLength += 6 - unicodeBuffer.byteLength;
              this.state = TokenizerStates.STRING_DEFAULT;
              continue;
            }
```

Reply:
LGTM